python3Packages.mayavi: 4.7.1 -> 4.7.2; vtk -> vtk_9

bcdarwin opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [NA] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

bcdarwin committed:
python3Packages.mayavi: 4.7.1 -> 4.7.2; vtk -> vtk_9

 - also removed python3Packages.vtk_{7,8} since nothing uses these and they're expensive to build

drewrisinger reviewed (approved):
* [x] Diff LGTM
* [x] Commits LGTM
* [x] Builds via ``nix-review``: Built everything through ``vtk_9``, but that was taking too long so I quit.

SuperSandro2000 commented on pkgs/top-level/python-packages.nix:
```diff
@@ -7341,10 +7341,10 @@ in {
   vsts-cd-manager = callPackage ../development/python-modules/vsts-cd-manager { };
 
-  vtk = toPythonModule (pkgs.vtk_7.override {
+  vtk_9 = disabledIf isPy27 (toPythonModule (pkgs.vtk_9.override {
```
Please move this into package. ``disabled = isPy2k``.

FRidh replied:
> Please move this into package. disabled = isPy2k.

This is only done with `buildPython*` packages, not other builders. Technically it could be done, but it would force a rebuild for the top-level package.

bcdarwin commented:
I've rebased onto master and resolved merge conflicts due to VTK expression refactoring.  I've left the `disabledIf isPy2k` for now.

I also removed vtk_7 and vtk_8 from the Python package set since nothing in the tree uses them and they're expensive to build.

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 97602` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>python37Packages.mayavi</li>
<li>python37Packages.vtk</li>
<li>python38Packages.mayavi</li>
<li>python38Packages.vtk</li>
<li>python39Packages.vtk</li>
</ul>
</details>

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 97602` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.mayavi: log was empty</li>
<li>python38Packages.mayavi: log was empty</li>
  </ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.vtk</li>
<li>python38Packages.vtk</li>
<li>python39Packages.vtk</li>
</ul>
</details>

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 97602` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.mayavi: log was empty</li>
<li>python38Packages.mayavi: log was empty</li>
  </ul>
</details>
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.vtk</li>
<li>python38Packages.vtk</li>
<li>python39Packages.vtk</li>
</ul>
</details>

```
error: --- Error ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- nixbuilder for '/nix/store/4wkaww74i5d5czvk8dcqqd5d0ir32ryy-python3.7-envisage-4.9.2.drv' failed with exit code 1; last 10 log lines:                                                            Ran 229 tests in 1.446s                                                                                                                                                                                                                                                                                                                                                                 FAILED (failures=19, errors=1, skipped=6)                                                                                                                                                   Error in atexit._run_exitfuncs:                                                                                                                                                             Traceback (most recent call last):                                                                                                                                                            File "/private/tmp/nix-build-python3.7-envisage-4.9.2.drv-0/envisage-4.9.2/envisage/plugins/ipython_kernel/kernelapp.py", line 198, in close                                                  self.close_shell()                                                                                                                                                                        File "/private/tmp/nix-build-python3.7-envisage-4.9.2.drv-0/envisage-4.9.2/envisage/plugins/ipython_kernel/kernelapp.py", line 215, in close_shell                                            shell = self.kernel.shell                                                                                                                                                               AttributeError: 'NoneType' object has no attribute 'shell'                                                                                                                                error: --- Error ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix1 dependencies of derivation '/nix/store/8k591n422x7f9h5jvddywsrgak6sd1g3-python3.7-mayavi-4.7.2.drv' failed to build                                                                       error: --- Error ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- nixbuilder for '/nix/store/6sx91w7mhxfxph4mp5ygbmq8w3n7qzvd-python3.8-envisage-4.9.2.drv' failed with exit code 1; last 10 log lines:                                                            Ran 229 tests in 1.434s                                                                                                                                                                                                                                                                                                                                                                 FAILED (failures=19, errors=1, skipped=6)                                                                                                                                                   Error in atexit._run_exitfuncs:                                                                                                                                                             Traceback (most recent call last):                                                                                                                                                            File "/private/tmp/nix-build-python3.8-envisage-4.9.2.drv-0/envisage-4.9.2/envisage/plugins/ipython_kernel/kernelapp.py", line 198, in close                                                  self.close_shell()                                                                                                                                                                        File "/private/tmp/nix-build-python3.8-envisage-4.9.2.drv-0/envisage-4.9.2/envisage/plugins/ipython_kernel/kernelapp.py", line 215, in close_shell                                            shell = self.kernel.shell                                                                                                                                                               AttributeError: 'NoneType' object has no attribute 'shell'                                                                                                                                error: --- Error ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix1 dependencies of derivation '/nix/store/zbxsz8garalqjhhm355nwdx6pnfbrsjl-python3.8-mayavi-4.7.2.drv' failed to build                                                                       error: --- Error ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- nix2 dependencies of derivation '/nix/store/xqmlhm3h4wq1d57bb6w9q7br097y2fad-env.drv' failed to build

```

bcdarwin commented:
Obsoleted by #133056.
